feature-libs/asm/customer-360/components/sections/asm-customer-customer-coupon/asm-customer-customer-coupon.component.ts

```diff
@@ -0,0 +1,206 @@
+import { CustomerCouponService } from '@spartacus/core';
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  combineLatest,
+  of,
+} from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { Customer360SectionContext } from '../customer-360-section-context.model';
+import { CustomerCouponEntry } from './asm-customer-customer-coupon.model';
+import { ICON_TYPE } from '@spartacus/storefront';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'cx-asm-customer-coupon',
+  templateUrl: './asm-customer-customer-coupon.component.html',
+})
+export class AsmCustomerCustomerCouponComponent implements OnInit, OnDestroy {
+  showErrorAlert$ = new BehaviorSubject<boolean>(false);
+  showErrorAlertForApplyAction$ = new BehaviorSubject<boolean>(false);
+  entries$: Observable<Array<CustomerCouponEntry>>;
+  subscription = new Subscription();
+  currentTabIsAssignable = true;
+  iconTypes = ICON_TYPE;
+  activeTab = 0;
+
+  constructor(
+    protected context: Customer360SectionContext<Customer360CustomerCouponList>,
+    protected customer360Facade: Customer360Facade,
+    protected customerCouponService: CustomerCouponService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.customerCouponService
+        .getClaimCustomerCouponResultError()
+        .subscribe((error) => {
+          if (error) {
+            this.changeTab(true);
+            this.showErrorAlertForApplyAction$.next(true);
+          }
+        })
+    );
+    this.subscription.add(
+      this.customerCouponService
+        .getDisclaimCustomerCouponResultError()
+        .subscribe((error) => {
+          if (error) {
+            this.changeTab(false);
+            this.showErrorAlertForApplyAction$.next(true);
+          }
+        })
+    );
+    this.fetchCustomerCoupons();
+    this.currentTabIsAssignable = true;
+    this.hideAllErrorAlert();
+  }
+
+  public fetchCustomerCoupons(): void {
+    this.entries$ = combineLatest([this.context.data$]).pipe(
+      map(([data]) => {
+        const entries: Array<CustomerCouponEntry> = [];
+        data.customerCoupons.forEach((customerCoupon) => {
+          entries.push({
+            code: customerCoupon.name,
+            name: customerCoupon.description,
+            codeForApplyAction: customerCoupon.code,
+            applied: false,
+          });
+        });
+        return entries;
+      }),
+      catchError(() => {
+        this.showErrorAlert$.next(true);
+        return of([]);
+      })
+    );
+  }
+
+  public changeTab(assignable: boolean): void {
+    this.currentTabIsAssignable = assignable;
+    this.hideAllErrorAlert();
+    this.entries$ = this.customer360Facade
+      .get360Data([
+        {
+          requestData: {
+            type: Customer360Type.CUSTOMER_COUPON_LIST,
+            additionalRequestParameters: {
+              assignable: assignable,
+              searchQuery: undefined,
```

Comment:
seem no need assignment `undefined` to var, Could you remove it ?

Reply:
removed